imap/autocreate.c

```diff
@@ -121,49 +120,51 @@ static int autocreate_sieve(const char *userid, const char *source_script)
         return 1;
     }
 
-    /* Check if sievedir is defined in imapd.conf */
-    if(!config_getstring(IMAPOPT_SIEVEDIR)) {
-        syslog(LOG_WARNING, "autocreate_sieve: sievedir option is not defined. Check imapd.conf");
-        return 1;
-    }
-
     /* Check if autocreate_sieve_compiledscript is defined in imapd.conf */
     if(!(compiled_source_script = config_getstring(IMAPOPT_AUTOCREATE_SIEVE_SCRIPT_COMPILED))) {
         syslog(LOG_WARNING, "autocreate_sieve: autocreate_sieve_compiledscript option is not defined. Compiling it");
         do_compile = 1;
     }
 
+    const char *sieve_script_dir;
```

Comment:
I still need to look at the rest of this properly, but in the meantime the Travis build is failing because this needs to not be `const`.

I guess it was never a problem before because we didn't bother freeing it?